mta/ext.go

```diff
@@ -54,23 +62,137 @@ func UnmarshalExt(content []byte) (*EXT, error) {
 	return &mtaExt, err
 }
 
-// mergeWithExtensionFiles merges the extensions in the sent order.
-// There are currently no validations performed on the ID, extends or schema version fields of the extensions.
-func mergeWithExtensionFiles(mta *MTA, extensions []string) error {
-	for _, extPath := range extensions {
-		mtaExtContent, err := fs.ReadFile(filepath.Join(extPath))
+func parseExtFile(extPath string) (*EXT, error) {
+	mtaExtContent, err := fs.ReadFile(filepath.Join(extPath))
+	if err != nil {
+		return nil, err
+	}
+	mtaExt, err := UnmarshalExt(mtaExtContent)
+	if err != nil {
+		return nil, errors.Wrapf(err, extUnmarshalErrorMsg, extPath)
+	}
+	return mtaExt, nil
+}
+
+type extensionDetails struct {
+	fileName string
+	ext      *EXT
+}
+
+// mergeWithExtensionFiles merges the extensions in the order of the 'extends' chain.
+// The extends chain, and the ID and schema version of each mtaext file is validated.
+func mergeWithExtensionFiles(mta *MTA, extensions []string, mtaPath string) error {
+	extensionsDetails, err := getSortedExtensions(extensions, mta.ID, mtaPath)
+	if err != nil {
+		return err
+	}
+
+	for _, extDetails := range extensionsDetails {
+		err = checkSchemaVersionMatches(mta, extDetails)
 		if err != nil {
 			return err
 		}
-		mtaExt, err := UnmarshalExt(mtaExtContent)
+		err = Merge(mta, extDetails.ext, extDetails.fileName)
```

Comment:
I can't, the merge is done in a loop. I only return an error in case an error occured.
Also, I changed the error type and added the file path - I think you didn't review the latest commit. Please check.